reposerver/repository/repository.go

```diff
@@ -1427,6 +1429,17 @@ func newEnv(q *apiclient.ManifestRequest, revision string) *v1alpha1.Env {
 	}
 }
 
+func addManagedByLabel(target *unstructured.Unstructured) {
+	labels := target.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string, 1)
+	}
+	if labels["app.kubernetes.io/managed-by"] == "" {
```

Comment:
In the case of Helm applications `"app.kubernetes.io/managed-by"` will be `Helm`, even though de-facto it is Argo CD that is doing the management. I'm not sure what the right thing to do is here tbh. Either Argo CD should overwrite it and set `argocd`, or leave as-is. 🤔

Reply:
I like what @lindhe suggested with making this opt-in. Other than making this PR a non-breaking change, anybody who uses this feature in the future will know what's going on because they will explicitly choose to use it.

This way, I feel like maintainers who want to be able to filter their k8s resources by ones managed by argo will want `managed-by: helm` to be overwritten with `argocd`.
Thoughts?

Reply:
Yes, exactly why I think we should have this as an opt-in! If we can offload this design choice to the users, it's all for the better.